File: io/avro/src/test/resources/compat/issue-754/src/java/LegacyAvroCaller.java
package io.bluetape4k.avro.compat.issue754.java;

import io.bluetape4k.avro.AvroReflectSerializer;

public final class LegacyAvroCaller {
    private LegacyAvroCaller() {
    }

    public static Object deserializeNull(AvroReflectSerializer serializer) {
        return serializer.deserialize(null, Object.class);
    }
}

File: io/avro/src/test/resources/compat/issue-754/src/kotlin/LegacyAvroCaller.kt
package io.bluetape4k.avro.compat.issue754.kotlin

import io.bluetape4k.avro.AvroReflectSerializer
import io.bluetape4k.avro.deserialize

fun restoreLegacyAvro(serializer: AvroReflectSerializer, payload: ByteArray): String? =
    serializer.deserialize(payload)
